parquet-column/src/main/java/org/apache/parquet/column/ParquetProperties.java

```diff
@@ -420,8 +429,27 @@ public Builder withDictionaryEncoding(String columnPath, boolean enableDictionar
       return this;
     }
 
-    public Builder withByteStreamSplitEncoding(boolean enableByteStreamSplit) {
-      this.enableByteStreamSplit = enableByteStreamSplit;
+    /**
+     * Enable or disable BYTE_STREAM_SPLIT encoding for FLOAT and DOUBLE columns.
+     *
+     * @param enable whether BYTE_STREAM_SPLIT encoding should be enabled
+     * @return this builder for method chaining.
+     */
+    public Builder withByteStreamSplitEncoding(boolean enable) {
+      this.enableByteStreamSplit = enable;
+      this.enableExtendedByteStreamSplit &= enable;
+      return this;
+    }
+
+    /**
+     * Enable or disable BYTE_STREAM_SPLIT encoding for FLOAT, DOUBLE, INT32, INT64 and FIXED_LEN_BYTE_ARRAY columns.
+     *
+     * @param enable whether BYTE_STREAM_SPLIT encoding should be enabled
+     * @return this builder for method chaining.
+     */
+    public Builder withExtendedByteStreamSplitEncoding(boolean enable) {
```

Comment:
WDYT about handling it like we do for dictionary encoding. Dictionary can be enabled/disabled globally or per specific columns. It would help experimenting.